Realm/ObjectServerTests/RLMSyncTestCase.m

```diff
@@ -195,6 +196,51 @@ - (RLMSyncUser *)logInUserForCredentials:(RLMSyncCredentials *)credentials
     return theUser;
 }
 
+- (RLMSyncUser *)makeAdminUser:(NSString *)userName password:(NSString *)password server:(NSURL *)url {
+    // Admin token user (only needs to be set up once ever per test run).
+    // Note: this is shared, persistent state between tests.
+    static RLMSyncUser *adminTokenUser = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        NSURL *adminTokenFileURL = [[RLMSyncTestCase rootRealmCocoaURL] URLByAppendingPathComponent:@"sync/admin_token.base64"];
+        NSString *adminToken = [NSString stringWithContentsOfURL:adminTokenFileURL
+                                                        encoding:NSUTF8StringEncoding error:nil];
+        RLMSyncCredentials *credentials = [RLMSyncCredentials credentialsWithAccessToken:adminToken
+                                                                                identity:[[NSUUID UUID] UUIDString]];
+        adminTokenUser = [self logInUserForCredentials:credentials server:url];
+    });
+    XCTAssertNotNil(adminTokenUser);
+
+    // Create a new user, which starts off without admin privileges.
+    RLMSyncCredentials *creds = [RLMSyncCredentials credentialsWithUsername:userName password:password register:YES];
+    RLMSyncUser *adminUser = [self logInUserForCredentials:creds server:url];
+    XCTAssertFalse(adminUser.isAdmin);
+
+    // FIXME: is there any way to do this without waiting X seconds?
+    // User should be created very quickly but not necessarily instantly.
+    [[NSRunLoop currentRunLoop] runUntilDate:[NSDate dateWithTimeIntervalSinceNow:5.0]];
```

Comment:
A 5 second delay every time this method is called isn't great. Can we instead wait for some event that signals the user has been created? (Doesn't the call to `-logInUserForCredentials:server:` already do that through its use of `waitForExpectationsWithTimeout:handler:` internally?)